src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.Expander.cs

```diff
@@ -865,7 +865,7 @@ private static bool IsPropertyNameOfObjectInitializer(SimpleNameSyntax identifie
 
                 while (parent != null)
                 {
-                    if (parent.Kind() == SyntaxKind.ObjectInitializerExpression)
+                    if (parent.IsKind(SyntaxKind.ObjectInitializerExpression, SyntaxKind.WithInitializerExpression))
```

Comment:
i'm fine with this.  however, actual question about `with`.  does it allow for nested initializers like object initializers do?

Reply:
`with` is limited to one level, simple assignments only.
- `{ Property = 1 }` is okay
- `{ [0] = 1 }` is illegal
- `{ 1, 2, 3 }` is illegal
- `{ Property = { ... } }` is illegal